Config/FilterSet.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/*
+ * This file is part of the `liip/LiipImagineBundle` project.
+ *
+ * (c) https://github.com/liip/LiipImagineBundle/graphs/contributors
+ *
+ * For the full copyright and license information, please view the LICENSE.md
+ * file that was distributed with this source code.
+ */
+
+namespace Liip\ImagineBundle\Config;
+
+use Liip\ImagineBundle\Exception\InvalidArgumentException;
+
+final class FilterSet implements FilterSetInterface
+{
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var string
+     */
+    private $dataLoader;
+
+    /**
+     * @var int
+     */
+    private $quality;
+
+    /**
+     * @var FilterInterface[]
+     */
+    private $filters = [];
+
+    /**
+     * @return string
+     */
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    /**
+     * @return string
+     */
+    public function getDataLoader(): string
+    {
+        return $this->dataLoader;
+    }
+
+    /**
+     * @return int
+     */
+    public function getQuality(): int
```

Comment:
this should have a phpdoc explaining what it is. (resp, as rob says the jpg quality and the png compression level values should have that doc, to explain the value range)